airbyte-cdk/python/airbyte_cdk/sources/streams/abstract_stream.py

```diff
@@ -0,0 +1,213 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import Any, Iterable, List, Mapping, Optional, Tuple, Union
+
+import requests
+from airbyte_cdk.models import AirbyteMessage, AirbyteStream, SyncMode
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.source import Source
+from airbyte_cdk.sources.utils import casing
+from airbyte_cdk.sources.utils.schema_helpers import InternalConfig
+from airbyte_cdk.sources.utils.slice_logger import SliceLogger
+from airbyte_cdk.sources.utils.transform import TransformConfig, TypeTransformer
+from airbyte_cdk.sources.utils.types import JsonType, StreamData
+
+
+class AbstractStream(ABC):
+    @abstractmethod
+    def read(
+        self,
+        cursor_field: Optional[List[str]],
+        logger: logging.Logger,
+        slice_logger: SliceLogger,
```

Comment:
what is the developer expected to do with `slice_logger`?

Reply:
oh, I just saw that this is pretty much preserving the signature used for the methods in `abstract_source`. I agree with maxime that ideally we do a facade if we can

Reply:
Updated to use a facade